lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -2144,7 +2145,8 @@ class SPIRVVectorExtractDynamic : public SPIRVInstruction {
     if (getValue(VectorId)->isForward())
       return;
     assert(getValueType(VectorId)->isTypeVector() ||
-           getValueType(VectorId)->isTypeJointMatrixINTEL());
+           getValueType(VectorId)->isTypeJointMatrixINTEL() ||
+           getValueType(VectorId)->isTypeCooperativeMatrixKHR());
```

Comment:
Unlike Intel extension - KHR doesn't allow Vector[Extract/Insert]Dynamic instruction to get an element of a matrix. Instead it introduces a separate mechanism via GEP + load/store. Bottomline, please don't change this assert.

Reply:
Applied